src/wp-includes/bookmark-template.php

```diff
@@ -102,13 +102,27 @@ function _walk_bookmarks( $bookmarks, $args = '' ) {
 			$title = ' title="' . $title . '"';
 		}
 		$rel = $bookmark->link_rel;
-		if ( '' !== $rel ) {
-			$rel = ' rel="' . esc_attr( $rel ) . '"';
-		}
+
 		$target = $bookmark->link_target;
 		if ( '' !== $target ) {
+			if ( is_string( $rel ) && '' !== $rel ) {
+				$rels   = explode( ' ', $rel );
+
+				if ( false === in_array( 'noopener', $rels, true ) ) {
+					$rels[] = 'noopener';
+				}
+				$rel    = implode( ' ', $rels );
```

Comment:
Micro-optimization: Dop the `implode()` only when adding `noopener` to the list of rels.
```suggestion
				if ( in_array( 'noopener', $rels, true ) ) {
					$rels[] = 'noopener';
					$rel     = implode( ' ', $rels );
				}
```